src/bundle.c

```diff
@@ -1109,35 +1141,23 @@ static int lex_sort(const void *a, const void *b)
 int list_local_bundles()
 {
 	char *path = NULL;
-	DIR *dir;
-	struct dirent *ent;
 	struct list *bundles = NULL;
 	struct list *item = NULL;
 
 	string_or_die(&path, "%s/%s", path_prefix, BUNDLES_DIR);
 
-	dir = opendir(path);
-	if (!dir) {
+	errno = 0;
+	bundles = get_dir_files_sorted(path);
+	if (errno) {
```

Comment:
Well in theory somebody could just remove the files in /usr/share/clear/bundles and this would raise a false positive error. The later code doesn't have problems if bundles is null so I figured under reporting a potential issue (which would be obvious as it wouldn't show anything) isn't a big deal. I'm happy to change this though.

Reply:
I agree with @bryteise here

Reply:
So I'm not familiar with the case where opendir would succeed and readdir or closedir would fail given the same stream pointer and not having closed it already. I am happy to improve this though. Resetting errno before calling readdir and verifying errno is 0 after (otherwise free'ing the list) and then checking the return of closedir and freeing the list in case it has a non-zero return should resolve and unfree'd memory on program exit.

Reply:
sounds good

Reply:
Fair enough, no reason to depend on the list functions not doing something weird in the future.